src/node/blockstorage.cpp

```diff
@@ -348,6 +331,11 @@ void BlockManager::Unload()
     }
 
     m_block_index.clear();
+
+    vinfoBlockFile.clear();
+    nLastBlockFile = 0;
+    setDirtyBlockIndex.clear();
+    setDirtyFileInfo.clear();
```

Comment:
In commit "Make blockstorage globals private members of BlockManager" (fad81b7a0f8ea300708321c3fcae7f75cade14ba)

Is this a pure refactor, or is there some change in behavior here? It seems like these lines are being moved from `UnloadBlockIndex` to `BlockManager::Unload` but neither function is directly calling the other.

It might be good to move this change to a separate commit before the "make private members" commit with an explanation about why it's safe. Not saying it's unsafe. It's just not obviously safe.

Reply:
`UnloadBlockIndex` calls `chainman.Unload()`, which again should call `BlockManager::Unload`. So only change should be the destruction order

Reply:
> `UnloadBlockIndex` calls `chainman.Unload()`, which again should call `BlockManager::Unload`. So only change should be the destruction order

So there is no code in some other place calling `BlockManager::Unload` and now having extra state cleared?

If you put this change in a separate commit and explicitly say whether it's _supposed_ to change behavior and maybe say how it doesn't change behavior, it can make it easier for future reviewers and future regression bugfixers, who I think would logically be asking these questions.

Reply:
Thanks Ryan. Reworked the pull into more commits with descriptions in the body.

Reply:
> Thanks Ryan.

Ok, you finally convinced me